core/embed/rust/src/micropython/buffer.rs

```diff
@@ -51,7 +74,18 @@ impl StrBuffer {
         if self.ptr.is_null() {
             &[]
         } else {
-            unsafe { slice::from_raw_parts(self.ptr, self.len) }
+            unsafe { slice::from_raw_parts(self.ptr.add(self.off.into()), self.len.into()) }
+        }
+    }
+
+    pub fn offset(&self, skip_bytes: usize) -> Self {
+        let off: u16 = unwrap!(skip_bytes.try_into());
+        assert!(off <= self.len);
+        assert!(self.as_ref().is_char_boundary(skip_bytes));
```

Comment:
i'd prefer `ensure!` for both, these look important enough to do in production.
then it should be safe to leave off `checked_*` below: `self.len - off` is definitely safe if `off <= len`, `self.off + off` could only overflow if `self.off + self.len` could overflow, and given that `off` only advances by as much as `len` decreases, that should not be possible either.

Reply:
`assert!` are checked in production, `ensure!` just adds fancy message, no?
Droped checked_ in 61244340b4d6a50adecb3e7c3313fb2fce53ffa3.